### ISSUE TITLE
SnapClient on Windows : Wasapi configurable dwChannelMask 

### ISSUE THREAD

**Issue by FredVDB** [badaix/snapcast#634]
Hello,

would it be possible, to pass through command line the dwChannelMask options, so other combinations than SPEAKER_FRONT_LEFT | SPEAKER_FRONT_RIGHT could be used ? My goal is to use 4 different SnapClient instances on Windows using a 7.1 soundcard.

Regards,
Frederik

**Comment by badaix** [comment]
is it possible in Windows to configure virtual stereo soundcards that map their two channels to channels of a physical device?

**Comment by FredVDB** [comment]
I did not find a software which was capable of doing so.

I managed to compile a custom Snapclient.exe as a proof of concept to test it out (by changing line 182 in wasapi_player.cpp ) to  waveformatExtended->dwChannelMask = SPEAKER_BACK_LEFT | SPEAKER_BACK_RIGHT;

However, audio quality (on a Windows 7 pc) was very poor, probably due to the shared mode. In exclusive mode audio was perfect, but then it was impossible to run multiple snapclients simultaneously (because the soundcard was unavailable to the other clients.

I'm trying to port the snapclient to c#, because I'm more familiar with .net code + I would try to use Bass.NET for audio (it's possible to do the channel mapping with their matrix mixer)
